mods/ra/maps/allies-08a/allies08a.lua

```diff
@@ -0,0 +1,148 @@
+--[[
+   Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+AlliedBoatReinforcements = { "dd", "dd" }
+TimerTicks = DateTime.Minutes(21)
+ObjectiveBuildings = { Chronosphere, AlliedTechCenter }
+ScientistType = { "chan", "chan", "chan", "chan" }
+ScientistFootprint = { CPos.New(28, 83), CPos.New(29, 83) }
+
+InitialAlliedReinforcements = function()
+	Trigger.AfterDelay(DateTime.Seconds(2), function()
+		Media.PlaySpeechNotification(greece, "ReinforcementsArrived")
+		Reinforcements.Reinforce(greece, { "mcv" }, { MCVEntry.Location, MCVStop.Location })
+		Reinforcements.Reinforce(greece, AlliedBoatReinforcements, { DDEntry.Location, DDStop.Location })
+	end)
+end
+
+CreateScientists = function()
+	scientists = Actor.Create(ScientistType, true, { Location = ScientistsExit.Location, Owner = greece })
```

Comment:
I don't think you can create multiple actors using this command, I suspect this is the issue. Try using a loop to create them, let them scatter and then save the references to the actors in a table. You can use said table then for the `Trigger` functions below.

Reply:
I'll have to do some learning over the weekend to tackle this. Not sure if this means anything, but I tried this with a single actor and got the same crash:

e1 275 is not revealed for player USSR (0)!
Fatal Lua Error: [string "BindingSupport.lua"]:30: Unable to convert parameter 0 to String
   at OpenRA.Scripting.ScriptContext.FatalError(String message)
   at OpenRA.Mods.Common.Scripting.TriggerGlobal.<>c__DisplayClass14.<OnEnteredFootprint>b__13(Actor a)
   at OpenRA.Mods.Common.Traits.ActorMap.CellTrigger.Tick(ActorMap actorMap)
   at OpenRA.Mods.Common.Traits.ActorMap.OpenRA.Traits.ITick.Tick(Actor self)
   at OpenRA.World.<Tick>b__d(TraitPair`1 x)
   at OpenRA.WorldUtils.DoTimed[T](IEnumerable`1 e, Action`1 a, String text)
   at OpenRA.World.Tick()
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.Program.Main(String[] args)

On a different note, I typically like to reward the player for completing an objective. Would chrono tank reinforcements be fitting, or would aftermath units be straying too far? (Thought about giving a new construction option for helicopters but I don't think that's doable atm)

Reply:
Try

```lua
scientist = Actor.Create("chan", true, { Location = ScientistsExit.Location, Owner = greece })
```

Reply:
Consider me baffled. I've tried Reinforcements.Reinforce instead of actor.Create; OnEnteredProximityTrigger instead of OnEnteredFootprint. Even tried a Tanya instead of Chan.
I see no difference in what I'm attempting here and how other missions use these types of triggers. (And i tested one to make sure it wasn't broken in bleed.) 
Crashes every time the footprint is entered:

Fatal Lua Error: [string "BindingSupport.lua"]:30: Unable to convert parameter 1 to String[]
   at OpenRA.Scripting.ScriptContext.FatalError(String message)
   at OpenRA.Mods.Common.Scripting.TriggerGlobal.<OnEnteredProximityTrigger>c__AnonStorey8.<>m__0(Actor a)
   at OpenRA.Mods.Common.Traits.ActorMap.ProximityTrigger.Tick(ActorMap am)
   at OpenRA.Mods.Common.Traits.ActorMap.OpenRA.Traits.ITick.Tick(Actor self)
   at OpenRA.World.<Tick>m__4(TraitPair`1 x)
   at OpenRA.WorldUtils.DoTimed[T](IEnumerable`1 e, Action`1 a, String text)
   at OpenRA.World.Tick()
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)
   at OpenRA.WindowsLauncher.Main(String[] args)

If we can't solve this, the only other idea I had was a "Repair the island facility" objective.

Reply:
Reinforcements have arrived: https://github.com/Smittytron/OpenRA/pull/1
PS: This needs a rebase.

Reply:
Updated, but I'm still getting a crash when the footprint trigger is activated. Now it's:

> Fatal Lua Error: [string "BindingSupport.lua"]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object.
>    at OpenRA.Mods.Common.Scripting.ReinforcementsGlobal.Reinforce(Player owner, String[] actorTypes, CPos[] entryPath, Int32 interval, LuaFunction actionFunc) in c:\Users\Smitty\OpenRA\OpenRA.Mods.Common\Scripting\Global\ReinforcementsGlobal.cs:line 80
>    --- End of inner exception stack trace ---
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
>    at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
>    at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
>    at OpenRA.Scripting.ScriptMemberWrapper.Invoke(LuaVararg args) in c:\Users\Smitty\OpenRA\OpenRA.Game\Scripting\ScriptMemberWrapper.cs:line 75
>    --- End of inner exception stack trace ---
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
>    at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
>    at Eluant.LuaRuntime.MethodWrapper.Invoke(Object[] parms)
>    at Eluant.LuaRuntime.MakeManagedCall(IntPtr state, MethodWrapper wrapper)
>    at OpenRA.Scripting.ScriptContext.FatalError(String message)
>    at OpenRA.Mods.Common.Scripting.TriggerGlobal.<>c__DisplayClass14.<OnEnteredFootprint>b__13(Actor a)
>    at OpenRA.Mods.Common.Traits.ActorMap.CellTrigger.Tick(ActorMap actorMap)
>    at OpenRA.Mods.Common.Traits.ActorMap.OpenRA.Traits.ITick.Tick(Actor self)
>    at OpenRA.World.<Tick>b__d(TraitPair`1 x)
>    at OpenRA.WorldUtils.DoTimed[T](IEnumerable`1 e, Action`1 a, String text)
>    at OpenRA.World.Tick()
>    at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
>    at OpenRA.Game.LogicTick()
>    at OpenRA.Game.Loop()
>    at OpenRA.Game.Run()
>    at OpenRA.Game.InitializeAndRun(String[] args)
>    at OpenRA.Program.Main(String[] args)
> 
>

Reply:
See https://github.com/OpenRA/OpenRA/pull/15582#discussion_r223170238.